src/irmin-pack/unix/file_manager_intf.ml

```diff
@@ -37,7 +37,10 @@ module type S = sig
   val dict : t -> Dict.t
   val suffix : t -> Suffix.t
   val index : t -> Index.t
-  val mapping : t -> Io.t option
+
+  val mapping : t -> string option
+  (** path to mapping file *)
```

Comment:
Do we need to expose this still? Since we can get this from the Layout?

Reply:
The semantics of this is "give me the mapping currently held by the file manager". As I understand it, the file manager is the entity that knows the "current" files that are in use. It is true that there is a connection with Layout and the file manager generation, but my understanding is that we should use the file manager to access the files, not try to circumvent it by accessing the files directly e.g. by finding their name from Layout. But my understanding could be wrong, and I'm happy to change this if you think it should be changed.

Reply:
I agree with @icristescu on this one. Since ext.ml relies on `Layout` to compute paths, I don't see a reason why dispatcher.ml shouldn't

Reply:
I spent an hour or so trying to remove this from the interface. The idea would be that the mapping file name is determined by the generation that is held by the File_manager. Unfortunately I could not get the resulting code to work. It seems that, probably during the swap (in the main process, after the child terminates), the generation may not be updated when the mapping file reloads, so the mapping loads the previous generation and something bad happens. Given the pressing need to get the mmap improvements into the codebase I propose that we keep this function for the time being, and eliminate it in a future PR.

Reply:
Let's do that